### PR TITLE
Tests: New `.testhtml` format to replace `.js` language tests

### PR DESCRIPTION
**Pull request by RunDevelopment**
I added a new test file format that replaces the `.js` language tests (all 5 of them).

**Motivation**

`.js` language tests are inconvenient because commands like `npm run test:languages -- --update` don't affect them. They also require more escaping because code is stored in JS strings.

A new test suite came up [here](https://github.com/JaKXz/prism/pull/1). Inspired by this, I came up with `.testhtml`. `.testhtml` files are very similar to our current `.test` files. The only difference is that they store the HTML output instead of the simplified JSON stream. This allows `.testhtml` files to test the same code as current `.js` files while being as convenient as `.test` files.

**Changes**

- Added `.testhtml` format (implementation + doc),
- Removed support for `.js` test files can converted all current `.js` test files into `.testhtml` files.
- Language tests can now load plugins. This means that languages can now test their compatibility/interaction with plugins.

---

/cc @JaKXz

### CODE REVIEW

**Comment by joshgoebel** [comment]
What makes these special and requiring a different format than the traditional `.test` format?

**Comment by RunDevelopment** [comment]
Two reasons:

1. `.test` files only included limited information about the token stream. E.g. they don't include aliases. `.js` files include the complete HTML produced by Prism.
2. The token streams stored in `.test` files are based on `Prism.tokenize` + a few hooks. This means that they only test [this section](https://github.com/PrismJS/prism/blob/a62ef7965dee348c22cbe7ee305de0125480ae1c/components/prism-core.js#L617-L624) of the highlighting process. On the other hand, `.js` files test almost the entire highlighting process.

Basically, `.test` files test everything _before_ a token stream gets converted into HTML while `.js` (and now `.testhtml`) files test (almost) everything.

**Comment by joshgoebel** [comment]
Thanks for the clarification.  I was curious about this because I've never loved how we do our markup testing... though the idea of just merging the input and output files has long been an idea I've had (ie, what you're doing with testhtml).  Currently we have them as entirely separate files in our repo.

**Review by JaKXz — COMMENTED** [review]
I think this is a step in a good direction! This is exactly a problem I was thinking about and glad that we're on the same page about the confidence in the system -- however, _with my limited context_ it seems like we're re-inventing a wheel for a solved problem...

I wrote a full response here: https://github.com/JaKXz/prism/pull/1#issuecomment-854862460

File: .gitattributes [hunk 1]
```diff
@@ -1,4 +1,5 @@
 * text=auto
 
 # Test files should not have their line endings modified by git
-/tests/languages/**/*.test binary
\ No newline at end of file
+/tests/languages/**/*.test binary
+/tests/languages/**/*.testhtml binary
```
[the remark this answers is not in the document]
Comment (RunDevelopment) [reply .gitattributes:5]: Line ends. Git will automatically checks out and commits line ends in whatever format is configured on your system. That's a problem for test files because the output now depends on your git config. We also want to be able to have tests that test code with different line ends.

[the remark this answers is not in the document]
Comment (joshgoebel) [reply .gitattributes:5]: > We also want to be able to have tests that test code with different line ends.

I'd suggest that's a poor reason to make ALL the tests `binary` though.  If the highlighter is expected to pass text thru with line endings preserved then surely several unit tests [coded in pure JS perhaps] (of various line encodings) could confirm this behavior and the behavior for the majority of the test suite could be to expect a common line encoding or to normalize the line encodings.

In working on a few projects where test files were purposely supposed to have weird encodings (or hidden space at the end of lines - when editors default to removing this) it's always been a huge pain.  Line endings and whitespace simply shouldn't break things.

[the remark this answers is not in the document]
Comment (RunDevelopment) [reply .gitattributes:5]: Frankly, you're right. We can just have the test file parser auto-convert line ends and that's it. We sorta require `\r\n` line ends for these files anyway.

That will be a fun PR. Changing the line ends in hunderts of files... Well, I'll do that tomorrow.

[the remark this answers is not in the document]
Comment (joshgoebel) [reply .gitattributes:5]: We recently went thru this sort of thing with [Wren](https://wren.io) as well (my new obsession).  The compiler/parser would treat different line endings very differently.  You'd mostly notice this inside multi-line strings where the value of the string could vary depending on how the file would saved... this was very counter intuitive - and we fixed it.

In my testing Ruby, Node.js, and Python all of them "normalize" the script input to a standardized line encoding (regardless of the on-disk encoding used)... so that whatever line encoding the script runs in the behavior is always consistent.  

> Changing the line ends in hunderts of files

Good luck!
